cluster-autoscaler/core/utils.go

```diff
@@ -174,6 +174,34 @@ func FilterOutExpendablePods(pods []*apiv1.Pod, expendablePodsPriorityCutoff int
 	return result
 }
 
+// FilterSchedulablePodsForNode filters pods that can be scheduled on the given node.
+func FilterSchedulablePodsForNode(context *context.AutoscalingContext, pods []*apiv1.Pod, nodeGroupId string, nodeInfo *schedulercache.NodeInfo) []*apiv1.Pod {
+	schedulablePods := []*apiv1.Pod{}
+	loggingQuota := glogx.PodsLoggingQuota()
+	podSchedulable := make(podSchedulableMap)
+	for _, pod := range pods {
+		schedulable, found := podSchedulable.get(pod)
+		if !found {
+			err := context.PredicateChecker.CheckPredicates(pod, nil, nodeInfo, simulator.ReturnVerboseError)
+			if err == nil {
+				schedulable = true
+				podSchedulable.set(pod, true)
+			} else {
+				glog.V(2).Infof("Predicate failed: %v", err)
+				schedulable = false
+				podSchedulable.set(pod, false)
+			}
+		}
+		if schedulable {
+			schedulablePods = append(schedulablePods, pod)
+		} else {
+			glogx.V(2).UpTo(loggingQuota).Infof("Pod %s can't be scheduled on %s", pod.Name, nodeGroupId)
```

Comment:
I don't like logging this line regardless if we actually run predicate or not. If we did run predicates for this pod it should be obvious from the log above (if it's not we should update that log). Also if we hit the limit some pods will have 2 lines ('Predicate failed' and 'cant be scheduled') and some will only have the former.

Maybe just change the `else` to `else if found` and update the log messages accordingly?

Reply:
done